src/improvements/doc/TASK_CREATION_GUIDE.md

```diff
@@ -0,0 +1,196 @@
+# Task Creation Guide
+
+This guide walks through the process of creating new tasks using existing templates, including network-specific considerations and best practices.
+
+## Quick Start
+
+To scaffold a new task using an existing template:
+
+```bash
+cd src/improvements/
+just new task
+```
+
+Follow the prompts from the justfile to create a new task directory with pre-populated README.md and config.toml files.
+
+## Directory Structure
+
+Tasks are organized by network in the following structure:
+
+```
+src/improvements/tasks/
+├── eth/           # Ethereum mainnet tasks
+├── sep/           # Sepolia testnet tasks
+├── oeth/          # Optimism Ethereum tasks
+├── opsep/         # Optimism Sepolia tasks
+└── common/        # Shared resources and documentation
+```
+
+### Network-Specific Organization
+
+1. Ethereum Mainnet (tasks/eth/)
+   - Production tasks
+   - Strict security requirements
+   - Full testing required
+   - Example: eth/001-security-council-phase-0/
+
+2. Sepolia Testnet (tasks/sep/)
+   - Testing and validation
+   - Development tasks
+   - Example: sep/001-op-extended-pause/
+
+3. Optimism Networks
+   - tasks/oeth/: Optimism Ethereum tasks
+   - tasks/opsep/: Optimism Sepolia tasks
+   - Network-specific configurations required
+
+### Task Naming Convention
+
+Tasks follow a numbered sequence within each network directory:
+- Format: `[network]/[###]-[descriptive-name]`
+- Example: `eth/001-security-council-phase-0`
+- Numbers help track task order and dependencies
+- Descriptive names should be clear and concise
+
+## Creating a New Task
+
+### 1. Use the Task Scaffolding Tool
+
+```bash
+cd src/improvements/
+just new task
+```
+
+The tool will:
+- Create a new numbered directory
+- Generate README.md template
+- Create config.toml template
+- Set up basic structure
+
+### 2. Configure the Task
+
+Edit config.toml to specify:
+
+1. Template Selection
+```toml
+templateName = "GasConfigTemplate"
+```
+
+2. L2 Chain Configuration
+```toml
+l2chains = [
+    {name = "Orderly", chainId = 291},
+    {name = "Metal", chainId = 1750}
+]
+```
+
+3. Template-Specific Parameters
+```toml
+[gasConfigs]
+gasLimits = [
+    {chainId = 291, gasLimit = 100000000},
+    {chainId = 1750, gasLimit = 100000000}
+]
+```
+
+### 3. Document the Task
+
+Update README.md with:
+- Task description
+- Requirements
+- Expected outcomes
+- Testing steps
+- Validation criteria
+- Status updates
+
+## Testing and Validation
+
+1. Local Testing
+```bash
```

Comment:
This is for testing the _template_ right? We are talking about user tasks here. I think we can delete this. 

Or maybe add in some commands like just simulate etc.